wifi-subsys/components/network/src/wifi.h

```diff
@@ -89,6 +89,76 @@ esp_err_t set_default_credentials();
  */
 uint8_t wifi_credentials_are_configured();
 
+/**
+ * @brief change the ssid of the sta and restart the wifi module
+ *
+ * @param sta_ssid [in] new ssid 
+ * @return esp_err_t ESP_OK: succeed, ESP_(others): fail 
+ */
+esp_err_t change_wifi_sta_ssid(char* sta_ssid);
+
+/**
+ * @brief change the ssid of the AP and restart the wifi module 
+ *         also save new ssid in the nvs
+ *
+ * @param sta_ssid [in] new AP ssid 
+ * @return esp_err_t ESP_OK: succeed, ESP_(others): fail 
+ */
+esp_err_t change_wifi_ap_ssid(char* ap_ssid);
+
+/**
+ * @brief change the password of the AP and restart the wifi module
+ *        also save new password in the nvs
+ *
+ * @param sta_ssid [in] new AP password 
+ * @return esp_err_t ESP_OK: succeed, ESP_(others): fail 
+ */
+esp_err_t change_wifi_ap_pass(char* pass_ap);
+
+/**
+ * @brief change the password of the STA and restart the wifi module
+ *        also save new password in the nvs
+ * 
+ * @param pass_sta [in] new sta password 
+ * @return esp_err_t ESP_OK: succeed, ESP_(others): fail
+ */
+esp_err_t change_wifi_sta_pass(char* pass_sta);
+
+/**
+ * @brief restart the wifi module 
+ *
+ * @return esp_err_t ESP_OK: succeed, ESP_(others): fail 
+ */
+esp_err_t wifi_restart();
+/**
+ * @brief turn off the WIFI module 
+ *
+ * @return esp_err_t ESP_OK: succeed, ESP_(others): fail 
+ */
+esp_err_t wifi_off();
+
+/**
+ * @brief is used when the default parameters still not save in storage, 
+ *        this verify that wifi mode is enabled in the default params and
+ *         will return the mode.
+ * 
+ * @return int 
+ */
+int identify_enabled_interface();
+/**
+ * @brief this function set the wifi mode and execute the connection,
+ *         it find the wifi mode in the nvs and it verify if already saved
+ *         the mode in the nvs. If the nvs return error or the wifi_mode is
+ *         equal to WIFI_MODE_NULL will take the defaults params for give 
+ *         warranty to the user that wifi init correctly this is very important
+ *         because if something failed  the wifi  will  stop.
+ * 
+ */
+esp_err_t set_wifi_mode();
```

Comment:
`set_wifi_mode` and `change_wifi_mode` sound similar,  I'm thinking  if both could be condensed in a single function

Reply:
Also as parameter the related `ESP_WIFI_MODE` enum could be used

Reply:
sounds similar but are two deferents  functions  `set_wifi_mode` is an reusable funtcions used many times insde in the Wifi module,  `change wifi mode` just receive the uart command and set new mode in the nvs and restart the wifi, maybe i should change for a better name.

Reply:
so one it's only for internal use and the other one as AT command :+1: